fix: Update cert instructions for NGINX One Console (#82)
nginx/documentation: content/nginx-one/how-to/config-sync-groups/add-file-csg.md

---

> docs: 
> doctypes:
>     - task
> tags:
>     - docs
> title: Add a file to a Config Sync Group
> toc: true
> weight: 400

Reviewer: ```suggestion
title: Add a file to a Config Sync Group
weight: 400
toc: true
type: how-to
product: NGINX One
docs: DOCS-000
```

Author: Does this matter -- in https://github.com/nginx/documentation/pull/90/files#diff-dfc68b874162469152f3646287019eae3428c3afb8bc5a75bb5becb5fe64b11cR1, I learned that some linters set up this indentation.

Reviewer: Yes, but I think that was merged before I had a chance to make an edit suggestion.

It's outside of scope for this PR: we can't impose linting rules on people until we have provided them an alternative.

It's a consistency/style concern, but not a structural engineering one that should block PRs, IMO.

---

## Overview

{{< include "nginx-one/add-file/overview.md" >}}

> 

Reviewer: ```suggestion

---

```

Author: @ADubhlaoich This should be lintable in some way... If you don't know how, I'll open an issue.

Reviewer: It's definitely lintable, we just don't have any tooling in place for it. It's something I can add with my upcoming work.

> ## Before you start

Reviewer: Flagging this as a general style issue I just noticed for the whole documentation set: we decided the heading was meant to be "Before you begin" ages ago.

https://github.com/nginx/documentation/blob/main/templates/how-to/template-how-to.md

Not sure how this ended up happening for NGINX One Console, should probably be fixed across the set as its own PR.

Author: Sounds like something else for a linter.

Author: ```suggestion
## Before you begin
```

Before you add files in your configuration, ensure:

- You have access to the NGINX One Console.
- Config Sync Groups are properly registered with NGINX One Console

> 

Reviewer: ```suggestion

---

```

## Important considerations

This page applies when you want to add a file to a Config Sync Group. Any changes you make here apply to all [Instances]({{< relref "/nginx-one/glossary.md" >}}) of that Config Sync Group.

> 

Reviewer: ```suggestion

---

```

## Add a file

You can use the NGINX One Console to add a file to a specific Config Sync Group. To do so:

1. Select the Config Sync Group to manage.
1. Select the **Configuration** tab.

   {{< tip >}}

   {{< include "nginx-one/add-file/edit-config-tip.md" >}}

   {{< /tip >}}

1. Select **Edit Configuration**.
1. In the **Edit Configuration** window that appears, select **Add File**.

You now have multiple options, described in the sections which follow.

> 

Reviewer: ```suggestion

---

```

### New Configuration File

Enter the name of the desired configuration file, such as `abc.conf` and select **Add**. The configuration file appears in the **Edit Configuration** window.

> 

Reviewer: ```suggestion

---

```

### New SSL Certificate or CA Bundle

{{< include "nginx-one/add-file/new-ssl-bundle.md" >}}

  {{< tip >}}

  Make sure to specify the path to your certificate in your NGINX configuration,
  with the `ssl_certificate` and `ssl_certificate_key` directives.

  {{< /tip >}}

> 

Reviewer: ```suggestion

---

```

### Existing SSL Certificate or CA Bundle

{{< include "nginx-one/add-file/existing-ssl-bundle.md" >}}
With this option, You can incorporate [Managed certificates]({{< relref "/nginx-one/how-to/certificates/manage-certificates.md#managed-and-unmanaged-certificates" >}}).

> 

Reviewer: ```suggestion

---

```

---
docs: 
doctypes:
    - task
tags:
    - docs
title: Add a file to a Config Sync Group
toc: true
weight: 400
---

## Overview

{{< include "nginx-one/add-file/overview.md" >}}

---

## Before you begin

Before you add files in your configuration, ensure:

- You have access to the NGINX One Console.
- Config Sync Groups are properly registered with NGINX One Console

---

## Important considerations

This page applies when you want to add a file to a Config Sync Group. Any changes you make here apply to all [Instances]({{< relref "/nginx-one/glossary.md" >}}) of that Config Sync Group.

---

> ## Add a file

Author: I've made / suggested corrections to both `content` pages. Since I call them (as variables) in the `add-file-csg.md` file, the corrections should replicate to [that page](https://frontdoor-test-docs.nginx.com/previews/docs/82/nginx-one/how-to/config-sync-groups/add-file-csg/) as well.

You can use the NGINX One Console to add a file to a specific Config Sync Group. To do so:

1. Select the Config Sync Group to manage.
1. Select the **Configuration** tab.

   {{< tip >}}

   {{< include "nginx-one/add-file/edit-config-tip.md" >}}

   {{< /tip >}}

1. Select **Edit Configuration**.
1. In the **Edit Configuration** window that appears, select **Add File**.

You now have multiple options, described in the sections which follow.

---

### New Configuration File

Enter the name of the desired configuration file, such as `abc.conf` and select **Add**. The configuration file appears in the **Edit Configuration** window.

---

### New SSL Certificate or CA Bundle

{{< include "nginx-one/add-file/new-ssl-bundle.md" >}}

  {{< tip >}}

  Make sure to specify the path to your certificate in your NGINX configuration,

>   with the `ssl_certificate` and `ssl_certificate_key` directives.

Reviewer 2: This is good for certificates and keys, but bundles have use cases.

The bundle file would be used for `ssl_trusted_certificate`, `ssl_client_certificate`, or `ssl_crl` directives. Maybe it would be good to link to the nginx.org docs? https://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_client_certificate

We probably want to double check if there other directives that can use cert or key files, or leave the suggestion here open ended.

Author: > We probably want to double check if there other directives that can use cert or key files, or leave the suggestion here open ended.

Good point. I see more in https://nginx.org/en/docs/http/ngx_http_ssl_module.html 

```suggestion
  with directives such as `ssl_certificate` and `ssl_certificate_key`. For a full list of related directives, see our documentation on the [ngx_http_ssl_module](https://nginx.org/en/docs/http/ngx_http_ssl_module.html).
```

  {{< /tip >}}

---

### Existing SSL Certificate or CA Bundle

{{< include "nginx-one/add-file/existing-ssl-bundle.md" >}}

> With this option, You can incorporate [Managed certificates]({{< relref "/nginx-one/how-to/certificates/manage-certificates.md#managed-and-unmanaged-certificates" >}}).

Author: Based on https://github.com/nginx/documentation/pull/82/files#r1934730807

```suggestion
```
